### DISCUSSION TITLE
Artisan command that generates interfaces

### DISCUSSION THREAD

**Discussion by Frank-Landry** [laravel/framework#40857] [Ideas]
Hello team.
My idea that I would like feedback on is this:
An artisan command that would help to generate interfaces.
For example:
``` php artisan make:interface SomeInterface```
and it would ask you a series of questions to help generate the interface like so:

1. Name of interface [name]
2. Location of new interface? [/some/example/path]
3. Would you like to add methods to interface? Y | N

If yes, loop and ask these questions, otherwise quit out and write the file and template:

1. Public, private or protected? [public]
2. Static? Y|N
3. Method name? [name]
4. Parameters in format of Type,Parametername? [String $name]
5. Return type? [type]
6. Add another method?

This is my first time asking about one of my ideas. I accept any and all feedback whether it's good or bad or constructive or mean.

**Comment by browner12** [comment]
What is the benefit over just creating the interface file?

**Reply by Frank-Landry** [reply]
The benefit is that it would be a tool to use in a scaffolding workflow and for those who prefer to use the cli to generate more of their code instead of having to type / copy paste it out. I've seen other frameworks with this kind of thing and thought it would be useful here. My other thought is if artisan can make controllers and models, why not make interfaces too? Please do not take this as me responding in a negative tone. I wish to convey a tone of explaining my reasoning behind the idea.

**Reply by gullyCZ** [reply]
> Jaká je výhoda oproti pouhému vytvoření souboru rozhraní?

For example, the one that creates the correct structure in which to place the interface file.

**Comment by ereztdev** [comment]
Hello @Frank-Landry That's a good thought process and I agree. See if you can follow these steps and then customize it to your liking. It gives you a thread to pull on.
https://medium.com/@ariadoos/laravel-custom-file-stubs-ed32f046ea81

**Comment by shaedrich** [comment]
Since this is supposed to be integrated with Laravel functionality, shouldn't the naming be "contract" instead of "interface", since that's the term, Laravel uses instead of the PHP one.
